### PR TITLE
rs: fix inherited classes, add tunnel endpoint update method

### PR DESCRIPTION
**Pull request by connor4312**
Noticed when implementing hosting in the Rust SDK that we lacked logic for dealing with inheritance.

### CODE REVIEW

**Review by jasongin — APPROVED** [review]

**Review by jasongin — DISMISSED** [review]

File: rs/src/management/http_client.rs [hunk 1]
```diff
@@ -234,21 +234,22 @@ impl TunnelManagementClient {
...
+    ) -> HttpResult<()> {
         let url = self.build_tunnel_uri(
             locator,
             Some(&format!("{}/{}", PORTS_API_SUB_PATH, port_number)),
         );
         let request = self
             .make_tunnel_request(Method::DELETE, url, options)
             .await?;
-        self.execute_json("create_tunnel_port", request).await
+        self.execute_json::<NeverDeserialize>("delete_tunnel_port", request).await?;
```
Comment (jasongin) [inline rs/src/management/http_client.rs:251]: The service returns 404 from a request to delete a port that doesn't exist. In the other SDKs that 404 is converted to a `false` return value. Does it make sense to do the same here? Also for `delete_tunnel()`.

File: rs/src/management/http_client.rs [hunk 2]
```diff
@@ -234,21 +234,22 @@ impl TunnelManagementClient {
         self.execute_json("create_tunnel_port", request).await
```
Comment (jasongin) [inline rs/src/management/http_client.rs:234]: This should be `update` instead of `create`.

File: rs/src/management/http_client.rs (hunk 1 above)
Comment (connor4312) [reply rs/src/management/http_client.rs:251]: This will return an HTTP error if the service gives back a 404

**Review by jasongin — APPROVED** [review]

**Comment by tanhakabir** [comment]
I believe my PR had already incorporated the changes here

**Pull request closed without merging**